thrust/examples/sum_columns.cu

```diff
@@ -0,0 +1,77 @@
+#include <thrust/equal.h>
+#include <thrust/execution_policy.h>
+#include <thrust/iterator/constant_iterator.h>
+#include <thrust/iterator/counting_iterator.h>
+#include <thrust/iterator/discard_iterator.h>
+#include <thrust/iterator/transform_iterator.h>
+#include <thrust/random.h>
+#include <thrust/reduce.h>
+#include <thrust/transform.h>
+#include <thrust/universal_vector.h>
+
+#include <cuda/std/mdspan>
+
+#include <iomanip>
+#include <iostream>
+
+int main()
+{
+  const int rows = 32;
+  const int cols = 16;
+
+  // Create a 2D multidimensional array of ints.
+  thrust::universal_vector<int> data(rows * cols, 42);
+  cuda::std::mdspan M(thrust::raw_pointer_cast(data.data()), rows, cols);
+
+  // Create a range to the linear index space for the multidimensional array.
+  auto idxs_begin = thrust::make_counting_iterator(0);
+  auto idxs_end   = idxs_begin + M.size();
+
+  // Fill the array with pseudorandom inputs in parallel on the device.
+  thrust::transform(thrust::device, idxs_begin, idxs_end, M.data_handle(), [] __host__ __device__(int ij) {
+    thrust::default_random_engine rng;
```

Comment:
Is it a problem that `rng` is created inside the lambda (resulting in identical state in every lambda invocation in parallel from all threads) and not captured externally?

Reply:
No, that's the point. We create it in each thread and then we `discard` by the linear index of the current element to get a unique PRNG state for each cell.

Reply:
got it. that's nice indeed